validator/src/main/java/com/google/daq/mqtt/util/ConfigDiffEngine.java

```diff
@@ -40,16 +37,24 @@ public List<String> computeChanges(Config deviceConfig) {
     return configUpdates;
   }
 
+  @SuppressWarnings("unchecked")
   private Map<String, Object> convertSemantics(Object thing) {
-    return thing == null ? ImmutableMap.of() :
-        Arrays.stream(thing.getClass().getFields())
-            .filter(field -> isNotNull(thing, field)).collect(
-                Collectors.toMap(Field::getName, field -> convertSemantics(thing, field)));
+    if (thing == null) {
+      return ImmutableMap.of();
+    }
+    if (thing instanceof Map) {
+      Map<String, Object> asMap = (Map<String, Object>) thing;
+      return asMap.keySet().stream()
+          .collect(Collectors.toMap(key -> key, key -> convertSemantics(asMap.get(key))));
+    }
+    return Arrays.stream(thing.getClass().getFields())
+        .filter(field -> isNotNull(thing, field)).collect(
```

Comment:
@grafnu this change appears to be causing a stack overflow on the endpoint reconfig tests and I can't figure it out -  https://github.com/noursaidi/udmi/actions/runs/3090008247/jobs/4998355934#step:5:13153